claripy/simplifications.py

```diff
@@ -195,6 +197,19 @@ def eq_simplifier(a, b):
         if a.op == 'BVV' and b.op != 'BVV':
             return b == a
 
+        # expr - c1 == c2    ->   expr == c1 + c2
+        if a.op == "__sub__" and len(a.args) == 2 and a.args[1].op == "BVV" and b.op == "BVV":
+            return a.args[0] == a.args[1] + b
+
+        # 1 ^ expr == 0     ->   expr == 1
+        if a.op == '__xor__' and b.op == "BVV" and b.args[0] == 0 and len(a.args) == 2:
+            if a.args[0].op == 'If' and a.args[1].op == "BVV" and a.args[1].args[0] == 1:   # If(expr, v1, v2) ^ 1 == 0
+                return a.args[0] == 1
```

Comment:
Why does the op being `If` matter? 

Also, the comment on 204 doesn't mathematically make sense; if `expr = 1` then `1^1 = 0`, which is invalid. In fact, `1^expr == 0` should always be `False`. The return statement on 207 therefore does not make sense to me.

Also, the comments on lines 204 and 206 conflict; the second seems right, and the comment on line 204 seems to conflict with what the code on 206 does.

Reply:
Oh, wait. `^` is XOR, not power.

Reply:
Though whether we want this is up to @ltfish.

Reply:
Addressed.